sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -778,5 +778,11 @@ private[sql] class SQLConf extends Serializable with CatalystConf with Logging {
   def clear(): Unit = {
     settings.clear()
   }
+
+  def reset(): Unit = {
+    sqlConfEntries.values.asScala.filter(_.isPublic).foreach { entry =>
```

Comment:
We can also change internal properties using `set` right? Why only reset public ones?